src/formatDate.js

```diff
@@ -8,7 +8,29 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat.pop();
+  const parts = date.split(separator);
+
+  const dateParts = {};
+
+  fromFormat.forEach((part, index) => {
+    dateParts[part] = parts[index];
+  });
+
+  if (toFormat.includes('YY') && dateParts['YYYY']) {
+    dateParts['YY'] = dateParts['YYYY'].slice(-2);
+  }
+
+  if (toFormat.includes('YYYY') && dateParts['YY']) {
+    dateParts['YYYY'] =
+      dateParts['YY'] < '30' ? '20' + dateParts['YY'] : '19' + dateParts['YY'];
+  }
+
+  const newSeparator = toFormat.pop();
```

Comment:
Similar to the issue with `fromFormat`, using `pop()` on `toFormat` modifies the original array. Consider using `slice()` or `spread operator` to avoid mutating the input array.